transmogrifier/sources/datacite.py

```diff
@@ -117,15 +118,39 @@ def get_optional_fields(self, xml: Tag) -> Optional[dict]:
 
         for date in xml.metadata.find_all("date"):
             d = timdex.Date()
-            if value := date.string:
-                if "/" in value:
-                    split = value.index("/")
-                    d.range = timdex.Date_Range(
-                        gte=value[:split],
-                        lte=value[split + 1 :],
-                    )
+            if date_value := date.string:
+                if "/" in date_value:
+                    split = date_value.index("/")
+                    gte_date = date_value[:split]
+                    lte_date = date_value[split + 1 :]
+                    if validate_date_range(
+                        gte_date,
+                        lte_date,
+                        source_record_id,
+                    ):
+                        d.range = timdex.Date_Range(
+                            gte=gte_date
+                            if validate_date(
```

Comment:
You don't need these additional checks for `validate_date` since `validate_date_range` does that already. Same in the other sources.